src/processor/tree/tree_wrapper.rs

```diff
@@ -79,35 +96,68 @@ impl TreeWrapper {
             hex::encode(root_hash)
         );
 
-        assert_eq!(root_hash.as_bytes(), block.new_state_root);
-        tracing::debug!("Successfully processed block {}", block.l2_block_number);
+        let root_hash_bytes = root_hash.as_bytes();
+        if root_hash_bytes == block.new_state_root {
+            tracing::debug!("Successfully processed block {}", block.l2_block_number);
+
+            Ok(root_hash)
+        } else {
+            tracing::error!(
+                "Root hash mismatch!\nLocal: {}\nPublished: {}",
+                hex::encode(root_hash_bytes),
+                hex::encode(block.new_state_root.clone())
```

Comment:
```suggestion
                hex::encode(&block.new_state_root)
```
Pretty sure there's no need to clone here

Reply:
the compiler disagrees :-)
```
$ cargo build
   Compiling state-reconstruct v0.1.0 (/home/vb/zksync-state-reconstruct)
error[E0507]: cannot move out of `block.new_state_root` which is behind a shared reference
   --> src/processor/tree/tree_wrapper.rs:108:29
    |
108 |                 hex::encode(block.new_state_root)
    |                             ^^^^^^^^^^^^^^^^^^^^ move occurs because `block.new_state_root` has type `Vec<u8>`, which does not implement the `Copy` trait

For more information about this error, try `rustc --explain E0507`.
error: could not compile `state-reconstruct` (bin "state-reconstruct") due to previous error
```

Reply:
Don't forget about `&` :wink: